requests_ntlm/requests_ntlm.py

```diff
@@ -180,3 +179,35 @@ def _auth_type_from_header(header):
         return 'Negotiate'
 
     return None
+
+def _get_server_cert(response):
+    """
+    Get the certificate at the request_url and return it as a SHA256 hash. Will get the raw socket from the
+    original response from the server. This socket is then checked if it is an SSL socket and then used to
+    get the hash of the certificate. The certificate hash is then used with NTLMv2 authentication for
+    Channel Binding Tokens support. If the raw object is not a urllib3 HTTPReponse (default with requests)
+    then no certificate will be returned.
+
+    :param response: The original 401 response from the server
+    :return: SHA256 hash of the DER encoded certificate at the request_url or None if not a HTTPS endpoint
+    """
+    certificate_hash = None
+    raw_response = response.raw
+
+    if isinstance(raw_response, HTTPResponse):
+        if sys.version_info > (3, 0):
+            socket = response.raw._fp.fp.raw._sock
+        else:
+            socket = response.raw._fp.fp._sock
+
+        if hasattr(socket, 'getpeercert') and callable(getattr(socket, 'getpeercert')):
```

Comment:
In fact, you can do this:

```python
getpeercert = getattr(socket, 'getpeercert', None)
if callable(getpeercert):
    server_certificate = getpeercert(True)
```

That should work pretty well, I think, and saves us doing too many repeatedly lookups into the same object.